### PR TITLE
[FEAT] Time/Stamina/Exp based skill boosts

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description

This PR implements the time/stamina/exp based skill boosts for the following skills:

- Cultivator
- Plant Whisperer
- Tree Hugger
- Coal Face
- Rush Hour
- Kitchen Hand

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Test suite

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/events/landExpansion/cook.ts [hunk 1]
```diff
@@ -17,6 +18,22 @@ type Options = {
   createdAt?: number;
 };
 
+type GetReadyAtArgs = {
+  item: ConsumableName;
+  skills: Partial<Record<BumpkinSkillName, number>>;
+  createdAt: number;
+};
+
+export const getReadyAt = ({ item, skills, createdAt }: GetReadyAtArgs) => {
```
Comment (adamhannigan) [inline src/features/game/events/landExpansion/cook.ts:27]: Do we need to use this inside of `cook()`?

File: src/features/game/events/landExpansion/plant.ts [hunk 2]
```diff
@@ -135,7 +174,13 @@ export function plant({
 
   expansion.plots = plots;
 
-  bumpkin.stamina.value -= PLANT_STAMINA_COST;
+  let staminaCost = PLANT_STAMINA_COST;
+
+  if (bumpkin.skills["Plant Whisperer"]) {
```
Comment (adamhannigan) [inline src/features/game/events/landExpansion/plant.ts:179]: Similar to the other boosts, should we extract this into a separate function for readability + testibility?

**Review by adamhannigan — APPROVED** [review]
LGTM 🏆 

It just looks like the cooking time boost is not being implemented yet inside of `cook.ts`

File: src/features/game/events/landExpansion/cook.ts (hunk 1 above)
Comment (spencerdezartsmith) [reply src/features/game/events/landExpansion/cook.ts:27]: 😅

**Review by adamhannigan — DISMISSED** [review]
LGTM

**Review by adamhannigan — APPROVED** [review]
LGTM ✅

**Pull request merged**